tests/travis/build-and-test

```diff
@@ -12,6 +12,12 @@ if [ -n "$CXX" ]; then
     QMAKE_ARGS="$QMAKE_ARGS QMAKE_CXX=$CXX"
 fi
 
+if [ "$CXX" = "clang++" ]; then
+    QMAKE_ARGS="$QMAKE_ARGS QMAKE_LINK=clang++"
+    QMAKE_ARGS="$QMAKE_ARGS QMAKE_LINK_SHLIB=clang++"
+    QMAKE_ARGS="$QMAKE_ARGS QMAKE_LFLAGS+=--coverage"
```

Comment:
We don't actually use the coverage for now, so we can stop using it during test runs.